### ISSUE TITLE
util/thread.cpp:217: [realm-core-2.1.0] pthread_mutex_lock() failed: Invalid mutex object provided

### ISSUE THREAD

**Issue by l-h-d** [realm/realm-cocoa#4846]
**Original title**: _is this issue resolved_

```
IMPORTANT: if you see this error, please send this log to help@realm.io.2017-04-10 17:12:29.906061[4209:1888252] util/thread.cpp:217: [realm-core-2.1.0] pthread_mutex_lock() failed: Invalid mutex object provided
0x0000000101b45a18 _ZN5realm4utilL18terminate_internalERNSt3__118basic_stringstreamIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 44
1                        0x0000000101b45c60 _ZN5realm4util9terminateEPKcS2_lOSt16initializer_listINS0_9PrintableEE + 368
2                        0x0000000101b4649c _ZN5realm4util5Mutex11lock_failedEi + 132
3                        0x0000000101b435bc _ZN5realm4util14remove_mappingEPvm + 352
4                        0x0000000101b443ac _ZN5realm4util6munmapEPvm + 28
5                        0x0000000101bfc58c _ZN5realm11SharedGroup5closeEv + 316
6                        0x0000000101bfcf9c _ZN5realm11SharedGroupD2Ev + 24
7                        0x0000000101a20c10 _ZN5realm5_impl16RealmCoordinatorD2Ev + 220
8   libc++.1.dylib                      0x0000000181900b5c _ZNSt3__119__shared_weak_count16__release_sharedEv + 60
9                           0x0000000101ad3640 _ZN5realm5RealmD2Ev + 68
10  libc++.1.dylib                      0x0000000181900b5c _ZNSt3__119__shared_weak_count16__release_sharedEv + 60
11  libobjc.A.dylib                     0x0000000181946f10 <redacted> + 148
12  libobjc.A.dylib                     0x00000001819537f4 objc_destructInstance + 92
13  CoreFoundation                      0x0000000182f185c8 <redacted> + 136
14                          0x0000000101ab2b38 -[RLMRealm dealloc] + 104
15  libobjc.A.dylib                     0x0000000181961fe0 <redacted> + 704
16  CoreFoundation                      0x0000000182dea028 _CFAutoreleasePoolPop + 28
17  UIKit                               0x0000000189049ae8 <redacted> + 596
18  UIKit                               0x00000001890497d4 <redacted> + 64
19  UIKit                               0x00000001895aafa4 <redacted> + 4484
20  CoreFoundation                      0x0000000182ebeb5c <redacted> + 24
21  CoreFoundation                      0x0000000182ebe4a4 <redacted> + 524
22  CoreFoundation                      0x0000000182ebc0a4 <redacted> + 804
23  CoreFoundation                      0x0000000182dea2b8 CFRunLoopRunSpecific + 444
24  GraphicsServices                    0x000000018489e198 GSEventRunModal + 180
25  UIKit                               0x0000000188e317fc <redacted> + 684
26  UIKit                               0x0000000188e2c534 UIApplicationMain + 208
27                           0x0000000100bed554 main + 124
28  libdyld.dylib                       0x0000000181dcd5b8 <redacted> + 4
```

how can i resolve this problem, can you help me? thanks

Steps：
1、Close network connection
2、Multiple query Realm and Initiating network requests
3、open the network link
4、Query again

I get the issue by this way, it maybe help you

**Comment by danlniel** [comment]
Can you tell us what steps to reproduce the issue?

**Comment by istx25** [comment]
Hi @l-h-d. We will need more information to be able to help you with this issue. Would you be able to go through the issue template (found here: https://github.com/realm/realm-cocoa/issues/new) and provide us with the most amount of information as possible. Thanks.

**Comment by istx25** [comment]
My apologies. @bdash is currently on vacation. Assigning to someone else.

**Comment by l-h-d** [comment]
Steps：
1、Close network connection
2、Multiple query Realm
3、open the network link
4、Query again

**Comment by TimOliver** [comment]
@l-h-d Just following up. Please update your issue with the information from our issue template so we can get a better idea of the issue you're experiencing. :)

**Comment by jpsim** [comment]
This was resolved in 2.1.1: https://github.com/realm/realm-cocoa/releases/tag/v2.1.1
